EngineLayer/NonSpecificEnzymeSearch/NonSpecificEnzymeSequencesToActualPeptides.cs

```diff
@@ -108,7 +108,9 @@ protected override MetaMorpheusEngineResults RunSpecific()
                                         if (index >= 0 && index >= digestionParam.MinPeptideLength)
                                         {
                                             //generate correct sequence
-                                            PeptideWithSetModifications tempPWSM = new PeptideWithSetModifications(pwsm, pwsm.OneBasedStartResidueInProtein, pwsm.OneBasedStartResidueInProtein + index - 1);
+                                            Dictionary<int, ModificationWithMass> allModsOneIsNTerminus = pwsm.AllModsOneIsNterminus
```

Comment:
the idea here is to just remove mods that can't be on the clipped peptide right? for your non specific search?

Reply:
wait I'm confused, is this a bug in mzLib? is it putting mods where it shouldn't? or is this some funky nonspecific thing

Reply:
Yes to the first. The second one is bug in mzlib, because digestion params isn't carried over in the constructor, which causes a parsimony crash now because multiprotease searches require them.